astropy/cosmology/flrw/tests/test_w0cdm.py

```diff
@@ -81,14 +83,12 @@ def test_clone_change_param(self, cosmo):
         # `w` params
         c = cosmo.clone(w0=0.1)
         assert c.w0 == 0.1
-        for n in set(cosmo.__parameters__) - {"w0"}:
-            v = getattr(c, n)
+        for n, v in filter_keys_from_items(c.parameters.items(), {"w0"}):
+            v_expect = getattr(cosmo, n)
             if v is None:
-                assert v is getattr(cosmo, n)
+                assert v is v_expect
             else:
-                assert u.allclose(
-                    v, getattr(cosmo, n), atol=1e-4 * getattr(v, "unit", 1)
-                )
+                assert u.allclose(v, v_expect, atol=1e-4 * getattr(v, "unit", 1))
```

Comment:
Use [assert_quantity_allclose()](https://docs.astropy.org/en/latest/api/astropy.tests.helper.assert_quantity_allclose.html) for better output if this tests ever fails. The tests shouldn't be using `assert u.allclose()` at all.

Reply:
✅